In [ ]:
import pandas as pd
import numpy as np
Data=pd.read_csv('/content/drive/MyDrive/Colab Notebooks/dataset w7.csv')

In [ ]:
Data.describe()

Total Marks
count    28.000000
mean     32.321429
std      12.454520
min      12.000000
25%      22.500000
50%      33.000000
75%      43.250000
max      50.000000

In [ ]:

import scipy.stats as stats
stats.ttest_ind(Data['Total Marks'][test_prep['Gender'] == 'male'],
                 Data['Total Marks'][Data['Gender'] == 'female'])

Ttest_indResult(statistic=-0.6940885268848668, pvalue=0.4937822317960763)

degrees of freedom = n-2 = 28-2 = 26
significance level = 5% = 0.05
corresponding t-score = 2.056

Here, |t-score|=0.69<2.056
Therefore, we cannot reject the null hypothesis.

Also, p-value=0.49>0.05
Therefore, we cannot reject the null hypothesis.

That is, the marks are not same.

In [ ]:

import scipy.stats as stats
stats.ttest_ind(Data['Total Marks'][Data['Test preparation'] == 'none'],Data['Total Marks'][Data['Test preparation'] == 'completed'])

Ttest_indResult(statistic=-3.6143637527769217, pvalue=0.0012671292457320106)

degrees of freedom = n-2 = 28-2 = 26
significance level = 5% = 0.05
corresponding t-score = 2.056

Here, |t-score|=3.614>2.056
Therefore, we can reject the null hypothesis.

Also, p-value=0.001<0.05
Therefore, we can reject the null hypothesis.

That is, Test preparation actually helps the students